src/Parsers/ParserAlterQuery.cpp

```diff
@@ -725,8 +731,24 @@ bool ParserAlterCommand::parseImpl(Pos & pos, ASTPtr & node, Expected & expected
                 if (!parser_modify_col_decl.parse(pos, command_col_decl, expected))
                     return false;
 
+                auto check_no_type = [&](const std::string_view keyword)
+                {
+                    if (!command_col_decl)
+                        return;
+
+                    const auto & column_decl = command_col_decl->as<const ASTColumnDeclaration &>();
+
+                    if (!column_decl.children.empty() || column_decl.null_modifier.has_value() || !column_decl.default_specifier.empty()
+                        || column_decl.ephemeral_default || column_decl.primary_key_specifier)
+                    {
+                        throw Exception(ErrorCodes::SYNTAX_ERROR, "Cannot specify column properties before '{}'", keyword);
+                    }
+                };
+
                 if (s_remove.ignore(pos, expected))
                 {
+                    check_no_type(s_remove.getName());
```

Comment:
Maybe we can do it the other way around and don't check for remove and so on if the type is declared. Then it will fail when it continues and tries to parse `REMOVE` later on. At least that's my theory. WDYT?

Reply:
My reasoning to do like this is because then we can give a very accurate error message to the user. As this is a breaking change in the syntax, I was aiming for helping the users as much as possible. I am happy to change the other way around, but then we lost the nice error message. Let me know if you think it is still better.

Maybe I can turn this lambda into a member function of `ASTColumnDeclaration`, so when somebody adds a new member, it will be harder to miss.

Reply:
I didn't think about the error message. I was thinking more about the parser not having that many issues with processing the query further.

Reply:
After discussing this on slack, we decided to go with the current version. If we run into any issues, we can change this.